src/components/Form/FormField.tsx

```diff
@@ -0,0 +1,207 @@
+
+    /** A control for the form field. */
+    control: customPropTypes.itemShorthand,
+
+    /**
+     * A custom render function for the control slot.
+     *
+     * @param {React.ReactType} Component - The computed component for this slot.
+     * @param {object} props - The computed props for this slot.
+     * @param {ReactNode|ReactNodeArray} children - The computed children for this slot.
+     */
+    renderControl: PropTypes.func,
+
+    /** The HTML input id. This will be set on the control element and will be use for linking it with the label for correct accessibility. */
+    id: PropTypes.string,
+
+    /** The HTML input name. */
+    name: PropTypes.string,
+
+    /** The HTML input type. */
+    type: PropTypes.string,
+
+    /** A label for the form field. */
+    label: customPropTypes.itemShorthand,
+
+    /**
+     * A custom render function for the label slot.
+     *
+     * @param {React.ReactType} Component - The computed component for this slot.
+     * @param {object} props - The computed props for this slot.
+     * @param {ReactNode|ReactNodeArray} children - The computed children for this slot.
+     */
+    renderLabel: PropTypes.func,
+
+    /** A field can have its label next to instead of above it. */
+    inline: PropTypes.bool,
+
+    /** A field can show that input is mandatory. */
+    required: PropTypes.bool,
+
+    /** Text message that will be displayed below the control (can be used for error, warning, success messages). */
+    message: customPropTypes.itemShorthand,
+
+    /**
+     * A custom render function for the message slot.
+     *
+     * @param {React.ReactType} Component - The computed component for this slot.
+     * @param {object} props - The computed props for this slot.
+     * @param {ReactNode|ReactNodeArray} children - The computed children for this slot.
+     */
+    renderMessage: PropTypes.func,
+
+    /** Additional CSS styles to apply to the component instance.  */
+    styles: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
+
+    /** Override for theme site variables to allow modifications of component styling via themes. */
+    variables: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
+  }
+
+  public static defaultProps = {
+    as: 'div',
+  }
+
+  public renderComponent({
+    ElementType,
+    classes,
+    accessibility,
+    variables,
+    styles,
+    rest,
+  }): React.ReactNode {
+    const {
+      children,
+      control,
+      renderControl,
+      label,
+      renderLabel,
+      id,
+      name,
+      type,
+      required,
+      message,
+      renderMessage,
+      controlType,
+    } = this.props
+
+    const labelElement = Text.create(label, {
+      defaultProps: {
+        as: 'label',
+        htmlFor: id,
+        styles: styles.label,
+      },
+      render: renderLabel,
+    })
+
+    const messageElement = Text.create(message, {
+      defaultProps: {
+        styles: styles.message,
+      },
+      render: renderMessage,
```

Comment:
just to be sure (but am pretty confident that this case is thought of) - it is safe to pass `null | undefined` value here, it will be just ignored, right?

Reply:
Yes, this is the case, I was testing this without sending the render method